### PR TITLE
feat: [CI-15410]: Updated code for V1 pipeline YAML

### PR DESCRIPTION
**Pull request by architmallik7**
Updated the convert.go for the new V1 pipeline YAML supporting the Run Step.

Tested the changes Locally:
Screenshot 2024-12-12 at 6 34 50 PM

### CODE REVIEW

File: convert/drone/convert.go [hunk 1]
```diff
@@ -165,23 +158,11 @@ func (d *Converter) convert(ctx *context) ([]byte, error) {
...
-					Steps:    convertSteps(from, d.orgSecrets),
-					Volumes:  convertVolumes(from.Volumes),
-
-					// TODO support for delegate.selectors from from.Node
-					// TODO support for stage.variables
-				},
+			pipeline.Stages = append(pipeline.Stages, &v2.StageV1{
+				Name:    from.Name,
+				Clone:   convertCloneV1(&from.Clone),
+				Runtime: "machine",
```
Comment (harshjain12) [inline convert/drone/convert.go:164]: We should not hard code it

File: convert/drone/convert.go [hunk 2]
```diff
@@ -236,28 +217,39 @@ func convertRegistry(src []*v1.Pipeline) *v2.Registry {
...
-		}
-	}
+func convertSteps(src *v1.Pipeline, orgSecrets []string) []*v2.StepV1 {
+	var dst []*v2.StepV1
 	for _, v := range src.Steps {
 		if v != nil {
 			switch {
 			case v.Detach:
-				dst = append(dst, convertBackground(v, orgSecrets))
+				continue
```
Comment (harshjain12) [inline convert/drone/convert.go:226]: we have to support this in next pr

**Review by harshjain12 — APPROVED** [review]

File: convert/drone/convert.go (hunk 1 above)
Comment (architmallik7) [reply convert/drone/convert.go]: updated the code and made Runtime dynamic.

File: convert/drone/convert.go (hunk 2 above)
Comment (architmallik7) [reply convert/drone/convert.go:234]: Sure

**Pull request merged**